src/main/java/com/microsoft/azure/functions/HttpStatusType.java

```diff
@@ -0,0 +1,71 @@
+package com.microsoft.azure.functions;
+
+/**
+ * Defines an HTTP Status Type
+ * 
+ * @since 1.0
+ */
+public interface HttpStatusType {
+
+    public static enum StatusClass {
+        INFORMATIONAL(1), SUCCESS(2), REDIRECTION(3), CLIENT_ERROR(4), SERVER_ERROR(5), OTHER(0);
+        int value = 0;
+
+        StatusClass(int _class) {
+            this.value = _class;
+        }
+
+        public static StatusClass statusClassOf(int code) {
+            int computed = code / 100;
+            for (StatusClass sc : StatusClass.values()) {
+                if (computed == sc.value) {
+                    return sc;
+                }
+            }
+
+            return OTHER;
+        }
+    }
+
+    public int getCode();
+
+    public String getReason();
```

Comment:
Remove getReason. 
Setting ReasonPhrase is hidden in AspNet ObjectResult https://github.com/aspnet/HttpAbstractions/issues/395. Will add support if there is an ask